src/screens/moviesScreen/MovieScreen.tsx

```diff
@@ -1,12 +1,80 @@
+import {useEffect, useState} from 'react';
+import {ScrollView} from 'react-native';
+
+import Carousel from '@components/carousel';
+import HorizontalMovieScroll from '@components/horizontalMovieScroll';
 import BaseContainer from '@components/baseContainer';
-import Movies from '@components/movies';
+import Loader from '@components/loader';
+
+import {discover} from '@network/apiUrls';
+import {fetchData} from '@network/apiMethods';
+
+import {useThemeColors} from '@theme/themes';
+import {styles} from './styles';
 
 const MoviesScreen = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [carouselData, setCarouselData] = useState([]);
+  const colors = useThemeColors();
+
+  const getData = async () => {
+    setIsLoading(true);
+    const movies = await fetchData(discover, {
+      params: {page: 1, sort_by: 'revenue.desc'},
+    });
+    const carouselMovies =
+      movies?.results?.map(
+        ({
+          id,
+          title,
+          poster_path,
+          overview,
+          vote_average,
+          vote_count,
+          release_date,
+        }: MovieCardData) => ({
+          id,
+          title,
+          poster_path,
+          overview,
+          vote_average,
+          vote_count,
+          release_date,
+        }),
+      ) || [];
+
+    setCarouselData(carouselMovies);
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    getData();
+  }, []);
+
+  if (isLoading) {
+    return <Loader />;
+  }
   return (
     <BaseContainer>
-      <Movies />
+      <ScrollView style={styles().container}>
+        <Carousel data={carouselData} />
+        <HorizontalMovieScroll
+          url={discover}
+          sortBy="popularity.desc"
+          sectionTitle="Popular movies"
+        />
+        <HorizontalMovieScroll
+          url={discover}
+          sortBy="vote_count.desc"
+          sectionTitle="Most viewed"
+        />
+        <HorizontalMovieScroll
+          url={discover}
+          sortBy="primary_release_date.asc"
+          sectionTitle="Old is Gold"
+        />
```

Comment:
use array approach-
It helps to maintain these many data and upcoming data easily, just add a new object and it will gets render, no need to write component again and again

const movieSections = [
  {
    sortBy: "popularity.desc",
    sectionTitle: "Popular movies",
  },
  {
    sortBy: "vote_count.desc",
    sectionTitle: "Most viewed",
  },
  {
    sortBy: "primary_release_date.asc",
    sectionTitle: "Old is Gold",
  },
];

movieSections.map(({ sortBy, sectionTitle }) => (
  <HorizontalMovieScroll
    key={sectionTitle}
    url={discover}
    sortBy={sortBy}
    sectionTitle={sectionTitle}
  />
));